test/groovy/CloudFoundryDeployTest.groovy

```diff
@@ -62,16 +66,24 @@ class CloudFoundryDeployTest extends BasePiperTest {
         .around(fileExistsRule)
         .around(dockerExecuteRule)
         .around(environmentRule)
-        .around(new JenkinsCredentialsRule(this).withCredentials('test_cfCredentialsId', 'test_cf', '********'))
+        .around(credentialsRule)
         .around(stepRule) // needs to be activated after dockerExecuteRule, otherwise executeDocker is not mocked
 
     @Before
     void init() {
+        rules =
         helper.registerAllowedMethod('influxWriteData', [Map.class], {m ->
             writeInfluxMap = m
         })
     }
 
+    @After
+    void cleanUp() {
+        // cleaning up credentials in case a test has modified
+        credentialsRule = new JenkinsCredentialsRule(this)
```

Comment:
I think this does not have any effect, as the old instance will still be in the RuleChain.

I think you have to rather rest the rule in init by keeping the instance.